components/performance-details/EditPerformanceModal.js

```diff
@@ -0,0 +1,284 @@
+import React, { useState, useEffect } from 'react'; // React
+import PropTypes from 'prop-types'; // PropTypes
+import axios from 'axios'; // axios
+import Navigation from '@containers/Navigation'; // Navigation state
+
+import Input from '@components/Input'; // Input
+import Modal from '@components/Modal'; // Modal
+import Dropdown, { formatDropdownOptions } from '@components/Dropdown'; // Dropdown
+import styles from '@styles/components/performances/PerformanceModal.module.scss'; // Component styles
+
+import { formatSchools } from '@utils/schools'; // Format schools util
+
+export default function EditPerformanceModal({
+  open,
+  setOpen,
+  setLoading,
+  getPerformance,
+  performance,
+}) {
+  const { event: eventId } = Navigation.useContainer();
+
+  const [danceTitle, setDanceTitle] = useState('');
+  const [dancersString, setDancersString] = useState('');
+  const [choreographersString, setChoreographersString] = useState('');
+  const [school, setSchool] = useState(null);
+  const [competitionLevel, setCompetitionLevel] = useState(null);
+  const [danceStyle, setDanceStyle] = useState(null);
+  const [danceSize, setDanceSize] = useState(null);
+
+  // Dropdown options
+  const [schoolDropdownOptions, setSchoolDropdownOptions] = useState([]);
+  const [performanceLevelDropdownOptions, setPerformanceLevelDropdownOptions] = useState([]);
+  const [danceStyleDropdownOptions, setDanceStyleDropdownOptions] = useState([]);
+  const [danceSizeDropdownOptions, setDanceSizeDropdownOptions] = useState([]);
+
+  const getFilters = async () => {
+    setLoading(true);
+
+    const getSchools = async () => {
+      try {
+        const response = await axios({
+          method: 'GET',
+          url: '/api/schools/collect',
+        });
+        const schools = formatSchools(response.data);
+
+        // Modal dropdown options
+        setSchoolDropdownOptions(
+          formatDropdownOptions(schools, {
+            value: 'id',
+            label: 'schoolName',
+          })
+        );
+      } catch {
+        // Empty catch block
+      }
+    };
+
+    const getSettings = async () => {
+      const response = await axios({
+        method: 'GET',
+        url: '/api/settings/collect',
+      });
+      const settings = response.data;
+
+      const performanceLevelSettings = settings.filter(
+        setting => setting.type === 'COMPETITION_LEVEL'
+      );
+      const danceStyleSettings = settings.filter(setting => setting.type === 'STYLE');
+      const danceSizeSettings = settings.filter(setting => setting.type === 'DANCE_SIZE');
+
+      const formatOptionsFields = {
+        value: 'id',
```

Comment:
I remember we had a bug surrounding what value and label were, can we put a comment somewhere around these so its clear what exactly id, value and label are referring to (especially since we have 2 'values' here.